File: OrderService/src/main/java/com/app/OrderService/DTO/Response/OrderResponse.java
package com.app.OrderService.DTO.Response;

import com.app.OrderService.DTO.BaseDTO.ItemEntity;
import com.app.OrderService.Enum.OrderStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class OrderResponse {
    String orderID;
    String customerID;
    String restaurantID;
    String shipperID;

    @Builder.Default
    List<ItemEntity> menu = new ArrayList<>();
    Long total;
    Instant time;
    OrderStatus status;
}

File: ReactTS/food-app-v2/src/APIs/dto/responses/OrderResponse.ts
import { ItemResponse } from "./ItemResponse";

export interface OrderResponse {
  orderID: string;
  customerID: string;
  restaurantID: string;
  shipperID: string | null;
  menu:ItemResponse[]
  total: number;
  time: string;
  status: string;
  address: string;
  phone: string;
}
